site/content/en/docs/concepts/workload_priority_class.md

```diff
@@ -0,0 +1,101 @@
+
+- want to prioritize workloads that remain inactive for a specific duration
+- want to set a lower priority for development workloads and higher priority for production workloads
+
+A WorkloadPriorityClass is a priorityClass which value is utilized by Kueue controller and 
+is independent from pod's priority.
+
+A sample WorkloadPriorityClass looks like the following:
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: WorkloadPriorityClass
+metadata:
+  name: sample-priority
+value: 10000
+description: "Sample priority"
+```
+
+## How to use WorkloadPriorityClass on Jobs
+
+The WorkloadPriorityClass is specified through a label `kueue.x-k8s.io/priority-class`.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: sample-job
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: sample-priority
+spec:
+...
+```
+
+The following workload is generated by the yaml above. The `PriorityClassName` field can accept either `PriorityClass` or `WorkloadPriorityClass` name as a value. To distinguish, when using `WorkloadPriorityClass`, a `priorityClassSource` field has the `kueue.x-k8s.io/workloadpriorityclass` value. When using `PriorityClass`, a `priorityClassSource` field has the `scheduling.k8s.io/priorityclass` value.
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: job-sample-job-7f173
+spec:
+  priorityClassSource: kueue.x-k8s.io/workloadpriorityclass
+  priorityClassName: sample-priority
+  priority: 10000
+  queueName: user-queue
+...
+```
+
+For other job frameworks, you can set `WorkloadPriorityClass` through a label same for Job.
+The Following is an example of `MPIJob`.
+
+```yaml
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: pi
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: sample-priority
+spec:
+...
+```
+
+## The relationship between pod's priority and workload's priority
+
+There are three scenarios for creating a workload from a job.
+1. A job specifies both `WorkloadPriorityClass` and `PriorityClass`
+- `WorkloadPriorityClass` is used for the workload's priority.
+- `PriorityClass` is used for the pod's priority.
+2. A job specifies only `WorkloadPriorityClass`
+- `WorkloadPriorityClass` is used for the workload's priority.
+- `WorkloadPriorityClass` is not used for pod's priority.
+3. A job specifies only `PriorityClass`
+- `PriorityClass` is used for the workload's priority and pod's priority.
+
+In certain job frameworks, there are CRDs that have multiple fields related to pod's priorities within their Spec, or pod priorities are defined in a different location compared to other jobs. In such scenarios, Kueue controller utilizes [`JobWithPriorityClass interface`](https://github.com/kubernetes-sigs/kueue/blob/e162f8508b503d20feb9b31fd0b27d91e58f2c2f/pkg/controller/jobframework/interface.go#L81-L84).  
```

Comment:
```suggestion
In certain job frameworks, there are CRDs that:
- Define multiple pod specs, where each can have their own pod priority, or
- Define the overall pod priority in a dedicated field.

In such scenarios, the integration of the CRD with Kueue utilizes the [`JobWithPriorityClass interface`](https://github.com/kubernetes-sigs/kueue/blob/e162f8508b503d20feb9b31fd0b27d91e58f2c2f/pkg/controller/jobframework/interface.go#L81-L84) to obtain the priority class. You can read the documentation for each job integration to learn how the priority class is obtained.
```

(do we have that information?)

Reply:
> You can read the documentation for each job integration to learn how the priority class is obtained.
(do we have that information?)

No, I couldn't find a good data source.
Instead, I wrote like this:
You can read the **code** for each job integration to learn how the priority class is obtained.